manila/share/drivers/netapp/dataontap/client/client_cmode.py

```diff
@@ -1859,12 +1859,32 @@ def modify_active_directory_security_service(
                 msg = _("Failed to modify existing CIFS server user-name. %s")
                 raise exception.NetAppException(msg % e.message)
 
+        if 'defaultadsite' in differring_keys:
+            if new_security_service['defaultadsite'] is not None:
+                cifs_server = self._get_cifs_server_name(vserver_name)
+                api_args = {
+                    'admin-username': new_security_service['user'],
+                    'admin-password': new_security_service['password'],
+                    'force-account-overwrite': 'true',
+                    'cifs-server': cifs_server,
+                    'default-site': new_security_service['defaultadsite']
+                }
+                try:
+                    LOG.debug("Trying to modify CIFS server with data: %s",
+                              api_args)
+                    self.send_request('cifs-server-modify', api_args)
+                except netapp_api.NaApiError as e:
+                    msg = _("Failed to modify CIFS server entry. %s")
+                    raise exception.NetAppException(msg % e.message)
+                self.configure_cifs_options(new_security_service)
+
         if 'server' in differring_keys:
             if current_security_service['server'] is not None:
                 self.remove_preferred_dcs(current_security_service)
 
             if new_security_service['server'] is not None:
                 self.set_preferred_dc(new_security_service)
+                self.configure_cifs_options(new_security_service)
```

Comment:
why is it necessary to configure the cifs option here? when it is done on line #1879

Reply:
if we change from server to defaultadsite, we have to modify server-discovery-mode